test_integration_lock_serializes has a race condition in exit-order assertion

## What happened

CI fails intermittently on \`test_integration_lock_serializes\` in \`tests/test_sprint_parallel.py\`:

\`\`\`
AssertionError: assert ['second', 'first'] == ['first', 'second']
\`\`\`

## What was expected

The test should reliably pass. The \`exited\` ordering assertion is not a property that \`integration_lock\` guarantees — "first" appends to \`exited\` *after* releasing the lock, so "second" can acquire the lock and append before "first" gets CPU time. The lock's actual serialization invariants (no overlap, FIFO entry order) are tested correctly by the \`overlap\` and \`entered\` assertions.

--- a/tests/test_sprint_parallel.py
+++ b/tests/test_sprint_parallel.py
@@ -2154,7 +2154,6 @@ class TestImmediateIntegrationLanding:
 
 def test_integration_lock_serializes(tmp_path: Path) -> None:
     entered: list[str] = []
-    exited: list[str] = []
     overlap = [False]
     inside = threading.Event()
     release = threading.Event()
@@ -2167,11 +2166,7 @@ def test_integration_lock_serializes(tmp_path: Path) -> None:
             entered.append(name)
             if name == "first":
                 release.wait(timeout=2)
-            else:
-                exited.append(name)
             inside.clear()
-        if name == "first":
-            exited.append(name)
 
     first = threading.Thread(target=worker, args=("first",))
     second = threading.Thread(target=worker, args=("second",))
@@ -2186,7 +2181,6 @@ def test_integration_lock_serializes(tmp_path: Path) -> None:
 
     assert overlap[0] is False
     assert entered == ["first", "second"]
-    assert exited == ["first", "second"]
 
 
 def test_write_sprint_summary_marks_cached_preflight(tmp_path):

